api/server/sdk/volume_ops.go

```diff
@@ -181,18 +182,45 @@ func (s *VolumeServer) create(
 			return "", err
 		}
 
-		newOwnership, updateNeeded := clone.Volume.Spec.GetCloneCreatorOwnership(ctx)
-		if updateNeeded {
-			// Set no authentication so that we can override the ownership
-			ctxNoAuth := context.Background()
+		// Generate update request based on input to create w/ parentID
+		updateReq := &api.SdkVolumeUpdateRequest{
+			VolumeId: id,
+		}
+		newOwnership, ownershipUpdateNeeded := clone.Volume.Spec.GetCloneCreatorOwnership(ctx)
+		if ownershipUpdateNeeded {
+			// Set no authentication so that we can override the ownership.
+			// TODO: this should be fixed to only remove auth metadata
+			ctx = context.Background()
+			updateReq.Spec = &api.VolumeSpecUpdate{
+				Ownership: newOwnership,
+			}
+		}
+
+		additionalLabelsNeeded := len(additionalCloneLabels) > 0
+		if additionalLabelsNeeded {
+			mergedLabels := make(map[string]string)
+			clonedLabels := make(map[string]string)
+			// Cloned volume may not have locator or volume.
+			// This is unlikely, but we'll avoid a panic here.
+			if clone.GetVolume() != nil && clone.GetVolume().GetLocator() != nil &&
+				clone.GetVolume().GetLocator().GetVolumeLabels() != nil {
```

Comment:
nit: the last check on line 206 is not needed since clonedLabels will just be nil on line 207 which is ok.